.github/workflows/cd-publishRelease.yml

```diff
@@ -0,0 +1,94 @@
+name: Publish Release
+
+on:
+    workflow_dispatch:
+      inputs:
+        target-branch:
+          description: 'Branch to target'
+          required: true
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+
+    steps:
+     - name: Checkout code
+       uses: actions/checkout@v3
+       with:
+        ref: ${{ github.event.inputs.target-branch }}
+        token: ${{ secrets.TAG_TOKEN }}
```

Comment:
Do we need these? Why are we taking the branch name here?

Reply:
We checkout to the required branch and then we can run the bal build (from inside the branch) which will build the jar file for the contents of that branch

Reply:
Check https://github.com/actions/checkout
When we select the branch from which this workflow should run, that'll be the ref. So need to specify this. Also we don't need the token to access this code.

Reply:
Yes, I have included the branch selection in the ref but if i didn't include the access token, it gives an authentication error🤔